### PR TITLE
Exclude draws

### PR DESCRIPTION
**Pull request by n-kall**
#### Summary

Adds an `exclude` option to `subset_draws` that will remove the matched draws. Addresses #333 

TODO:

~~Tests~~

#### Copyright and Licensing

By submitting this pull request, the copyright holder is agreeing to 
license the submitted work under the following licenses:

- Code: BSD 3-clause (https://opensource.org/licenses/BSD-3-Clause)
- Documentation: CC-BY 4.0 (https://creativecommons.org/licenses/by/4.0/)

### CODE REVIEW

**Comment by n-kall** [comment]
I tried to change as little code as necessary to implement this. But I think this is now ready for review

File: R/subset_draws.R [hunk 1]
```diff
@@ -55,6 +61,22 @@ subset_draws.draws_matrix <- function(x, variable = NULL, iteration = NULL,
   iteration <- check_iteration_ids(iteration, x, unique = unique)
   chain <- check_chain_ids(chain, x, unique = unique)
   draw <- check_draw_ids(draw, x, unique = unique)
+
+  if (exclude) {
```
Comment (paul-buerkner) [inline R/subset_draws.R:65]: please input check exclude via as_one_logical

**Review by paul-buerkner — CHANGES_REQUESTED** [review]
Thank you for working on this. I have left a few rather small comments I think

File: R/subset_draws.R (hunk 1 above)
Comment (n-kall) [reply R/subset_draws.R]: Thanks! Keep forgetting to do this. I now added input checking into the check_* functions, as is done with the the other arguments `regex` and `unique`. This does mean that the inputs are not checked if all `draw`, `iteration`, `variable`, `chain` are NULL though

[the remark this answers is not in the document]
Comment (n-kall) [reply R/subset_draws.R]: Good point, I changed this

**Comment by n-kall** [comment]
@paul-buerkner I think I addressed your comments now, also added more tests

**Pull request merged**